pkg/httpserver/ndm/http_server.go

```diff
@@ -0,0 +1,65 @@
+package ndm
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openebs/node-disk-manager/pkg/httpserver"
+	//"k8s.io/klog/glog"
+)
+
+func StartHttpServer() {
+
+	go func() {
+		err := http.ListenAndServe(":8080", nil)
+
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	go func() {
+		svc_ip := os.Getenv("NDO_OPERATOR_SERVICE_IP")
```

Comment:
This creates a hard dependency. between the NDM operator and NDM DS. Also poses a challenge in the installers for passing the NDO service IP to the NDM DS. 

Instead - can the NDM daemon set update the Device CRs with the status ( liveness timestamp at regular intervals) The NDO can look at that device CRs last timestamp and then decide to mark them inactive. And later when the NDM daemon set comes up, it anyway marks the device with the active status.

Reply:
I agree with you. Is it okay if we provide service_ip via cmdline argument to NDM ? 

Related to approach of updating CR, it seems to be reasonable and need a separate thread in NDM-Daemonset which periodically update Disk-CR with latest timestamp. But sometime it is too much work to etcd (specially when there are 100s of disks per node) so we did not go with this approach. Let sync-up again on liveness approach with other folks(who helped to evolve current implementation of liveness check) too to check if they can help us for a better approach.